crowbar_framework/app/models/provisioner_service.rb

```diff
@@ -242,22 +233,21 @@ def enable_repository(platform, arch, repo, bag = nil)
     @logger.debug("ID for #{repo} is #{repo_id}") if repo_id != repo
 
     if repo_object.available?
-      bag[platform] ||= {}
-      bag[platform][arch] ||= {}
-      repo_hash = repo_object.to_databag_hash
-      if bag[platform][arch][repo_id] != repo_object.to_databag_hash
+      repo_in_db = repo_object.data_bag_item
+      repo_current = repo_object.to_databag
+      if repo_in_db.nil? || Crowbar::Repository.data_bag_item_to_hash(repo_in_db) != repo_current.to_hash
```

Comment:
Metrics/LineLength: Line is too long. [105/100] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)